src/async_callback.rs

```diff
@@ -0,0 +1,577 @@
+            }
+        })
+    }
+
+    /// Finish the async sequence by calling the given Lua function with the arguments as the
+    /// current contents of the stack.
+    pub fn tail_call(self, func: &LocalFunction<'seq>) {
+        visit_shared(move |shared| {
+            *shared.next = SequencePoll::TailCall {
+                function: func.fetch(shared.locals),
+            };
+        });
+    }
+
+    /// Yield to the calling code (or to `to_thread`) values starting at `bottom` in the stack. When
+    /// this `Sequence` is resumed, resume arguments will be placed at `bottom` in the stack.
+    pub async fn yield_to(
+        &mut self,
+        to_thread: Option<&LocalThread<'seq>>,
+        bottom: usize,
+    ) -> Result<(), LocalError<'seq>> {
+        visit_shared(move |shared| {
+            *shared.next = SequencePoll::Yield {
+                to_thread: to_thread.map(|t| t.fetch(shared.locals)),
+                bottom,
+            };
+        });
+        wait_once().await;
+        visit_shared(move |shared| {
+            if let Some(err) = shared.error.take() {
+                Err(Local::stash(&shared.ctx, shared.locals, err))
+            } else {
+                Ok(())
+            }
+        })
+    }
+
+    /// Finish the async sequence by yeilding the current contents of the stack to the caller (or
+    /// to `to_thread`).
+    pub fn tail_yield(self, to_thread: Option<&LocalThread<'seq>>) {
+        visit_shared(move |shared| {
+            *shared.next = SequencePoll::TailYield {
+                to_thread: to_thread.map(|t| t.fetch(shared.locals)),
+            };
+        });
+    }
+
+    /// Resume `thread` with arguments starting at `bottom` in the stack. When the thread completes,
+    /// return values will be placed at `bottom` in the stack.
+    pub async fn resume(
+        &mut self,
+        thread: &LocalThread<'seq>,
+        bottom: usize,
+    ) -> Result<(), LocalError<'seq>> {
+        visit_shared(move |shared| {
+            *shared.next = SequencePoll::Resume {
+                thread: thread.fetch(shared.locals),
+                bottom,
+            };
+        });
+        wait_once().await;
+        visit_shared(move |shared| {
+            if let Some(err) = shared.error.take() {
+                Err(Local::stash(&shared.ctx, shared.locals, err))
+            } else {
+                Ok(())
+            }
+        })
+    }
+
+    /// Finish the async sequence by resuming `thread`, using the current contents of the stack
+    /// as arguments.
+    pub fn tail_resume(self, thread: &LocalThread<'seq>) {
+        visit_shared(move |shared| {
+            *shared.next = SequencePoll::TailResume {
+                thread: thread.fetch(shared.locals),
+            };
+        });
+    }
+}
+
+/// A collection of stashed values that are local to a specific `AsyncSequence`.
+///
+/// `Local` values are branded by `'seq` and cannot escape their parent `AsyncSequence` and are (for
```

Comment:
You may want to add [reference links](https://doc.rust-lang.org/rustdoc/write-documentation/linking-to-items-by-name.html) for some of these to make the docs easier to navigate.  ``[`Local`]`` should be parsed as a link to the `Local` struct, since it's in-scope; in general, normal Rust path syntax should work, and `cargo doc` will warn on unresolved links.